mlir/utils/jenkins/Jenkinsfile

```diff
@@ -793,5 +812,72 @@ pipeline {
                 }
             }
         }
+        stage ("MIGraphX") {
+            when {
+                beforeAgent true;
+                equals expected: true, actual: params.checkMIGraphX;
+            }
+            matrix {
+                axes {
+                    axis {
+                        name 'CODEPATH'
+                        values 'mfma', 'vanilla', 'navi21'
+                    }
+                }
+                agent {
+                    docker {
+                        image dockerImage()
+                        args dockerArgs()
+                        label getLabelFromCodepath("${CODEPATH}")
+                        alwaysPull true
+                    }
+                }
+                when {
+                    beforeAgent true
+                    expression { return shouldRunFromCodepath("$CODEPATH") }
+                }
+                environment {
+                    HOME="${WORKSPACE}"
+                }
+                stages {
+                    stage('Environment') {
+                        steps {
+                            echo "codepath is ${CODEPATH}"
+                            showEnv()
+                        }
+                    }
+                    stage("Install MIGraphX Dependencies") {
+                        steps {
+                            sh 'apt-get install miopen-hip'
+                            sh 'cget -p /MIGraphXDeps install ${WORKSPACE} -DBUILD_MIXR_TARGET=On -DLLVM_ENABLE_ZSTD=Off -DLLVM_ENABLE_THREADS=Off'
+                        }
+                    }
+                    stage("Build MIGraphX with MLIR") {
+                        steps {
+                            sh 'rm -rf MIGraphX'
+                            dir('MIGraphX') {
+                                getAndBuildMIGraphX("-DCMAKE_PREFIX_PATH=/MIGraphXDeps -DMIGRAPHX_ENABLE_MLIR=On -DCMAKE_CXX_COMPILER=/opt/rocm/llvm/bin/clang++")
+                            }
+                        }
+                    }
+                    stage("Verify MIGraphX with MLIR") {
+                        steps {
+                            dir('MIGraphX/build') {
```

Comment:
Can we run the unit tests for MIGraphX as well?

Reply:
We can but not all of them are sensitive our changes -- if anything we should only run test_gpu_mlir for which I dont know a way to carve that out.

Reply:
(I can take look if there is something LIT_FILTER)

Reply:
Fair enough. 

Also, do we know if the validation will for sure go through MLIR?

Reply:
-DMIGRAPHX_ENABLE_MLIR=On is the guarantee.

Reply:
(Happy to know if there is an alternative way make sure)

Reply:
OK, I've added the running mlir unit tests.